src/Simplex/Chat/Messages.hs

```diff
@@ -595,6 +595,27 @@ ciFileLoaded = \case
   CIFSRcvWarning {} -> False
   CIFSInvalid {} -> False
 
+data ForwardFileError = FFENotAccepted FileTransferId | FFEInProgress | FFEFailed | FFEMissing
+  deriving (Eq, Ord)
+
+ciFileForwardError :: FileTransferId -> CIFileStatus d -> Maybe ForwardFileError
```

Comment:
to not duplicate logic ciFileLoaded function could be defined via this one (isJust, seems like it's full match)

Reply:
ciFileLoaded can now be removed, it's unused